Manipulando streams

AndreSchuh opened the pull request.

Commit:
Exemplo de streams

AndreSchuh committed:
Desenvolvendo o exercicio de streams

prixua commented on src/main/java/br/andre/estudos/streams/StreamsBigDecimalExemplo.java:
```diff
@@ -0,0 +1,110 @@
+        System.out.println("Números 30 a 60 com 30% de desconto: ");
+
+        numeros.stream()
+                .filter(n -> n.compareTo(BigDecimal.valueOf(60)) <= 30)
```
O compareTo retorna apenas 3 valores, -1, 0 ou 1.

Veja a documentação 

https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/math/BigDecimal.html#compareTo(java.math.BigDecimal)

![image](https://github.com/user-attachments/assets/8e8b4b48-dedd-4392-ad89-9e6be934d222)


Logo vc não pode usar a comparação  `<=30`, pois fica muito estranho.
Verifica os outros métodos que também possam estar usando dessa forma.

prixua reviewed (commented):
Verificar os comentários

Commit:
- Mudei o método que gera os números para retornar números fixos para ficar mais fácil de comparar.
- Comentei os filtros para ficar mais fácil de ver os exemplos.
- Mudei para uma estrutura de List, para os números ficarem ordenados e ficar mais fácil de acompanhar.

AndreSchuh committed:
Atualizando

AndreSchuh committed:
Testes das classes feitos e atualização das classes para deixá-las testáveis.

AndreSchuh committed:
Melhorando os testes e adicionando exemplo com parametrized test

Commit:
- Alterado teste para usar arredondamento e fazer assert sem usar `for`
